### PR TITLE
Added a support for assume role call in AWS EC2 script.

### PR DESCRIPTION
**Pull request by PradeepSurale**
- Added a support for the call of AWS Assume Role.
- Now, we have support for both with direct user permission or by switching role to perform different operation.
- To make a use of Assume role call, we have added three properties which is:
  1. assume-role-arn : ARN of the role to be assumed.
  2. assume-role-timeout : Session timeout for keep session alive. ( min 900 sec to max 3600 sec)
  3. assume-role-session-name : A name for the session.
- If we are using the switch role functionality, that time for script the assume-role-arn property is mandatory, and value for other two i.e.  assume-role-timeout and  assume-role-session-name is not passed it will pick the default 3600 seconds and current time correspondingly. 
- Usage example, lets say we want to create cluster.
`-./snappy-ec2 -k <your-key-name> -i <your-keyfile-path> stop snap_ec2_cluster --with-zeppelin --authorized-address=103.243.236.11/32 --assume-role-arn=<role-arn> --assume-role-timeout=<timeout> --assume-role-session-name=<name-for-session>`

### CODE REVIEW

File: aws/ec2/snappy_ec2.py [hunk 1]
```diff
@@ -1527,11 +1539,30 @@ def real_main():
...
+            else:
+                conn = boto.ec2.connect_to_region(opts.region, aws_access_key_id=response['Credentials']['AccessKeyId'], 
+                        aws_secret_access_key=response['Credentials']['SecretAccessKey'],
+                        security_token=response['Credentials']['SessionToken'])
         else:
-            conn = ec2.connect_to_region(opts.region, profile_name=opts.profile)
+            if opts.assume_role_arn is None:
+                conn = ec2.connect_to_region(opts.region, profile_name=opts.profile)
+            else:
+                conn = boto.ec2.connect_to_region(opts.region, profile_name=opts.profile,
```
Comment (ashetkar) [inline aws/ec2/snappy_ec2.py]: Are there any other places where we need to put the assumed-role's credentials? Or is this is the only place?
Are the commands start, stop, etc also taken care of apart from the launch and destroy?

Comment (PradeepSurale) [reply aws/ec2/snappy_ec2.py]: I have tested the start and stop cluster commands it working fine. I think this is the only place need assume_role call.

File: aws/ec2/snappy_ec2.py [hunk 2]
```diff
@@ -331,6 +333,16 @@ def parse_args():
     parser.add_option(
         "--instance-profile-name", default=None,
         help="IAM profile name to launch instances under")
+    parser.add_option(
+        "--assume-role-arn", type="string", default=None,
+        help="ARN of assumed role to launch a instance")
```
Comment (ashetkar) [inline aws/ec2/snappy_ec2.py]: A more elaborate help would help. e..g "ARN of the IAM role to assume. This IAM role's credentials will be used to launch the cluster."

File: aws/ec2/snappy_ec2.py [hunk 3]
```diff
@@ -331,6 +333,16 @@ def parse_args():
...
+    parser.add_option(
+        "--assume-role-timeout", type="int", default=3600,
+        help="Session timeout of the assumed role, min is 900 seconds and max is 3600 seconds")
```
Comment (ashetkar) [inline aws/ec2/snappy_ec2.py]: Suggested help text: "Timeout in seconds for the temporary credentials of the assumed IAM role."

BTW, is the limit (900 - 3600) enforced?

File: aws/ec2/snappy_ec2.py [hunk 4]
```diff
@@ -331,6 +333,16 @@ def parse_args():
...
+    parser.add_option(
+        "--assume-role-session-name", type="string", default=datetime.now().strftime("%m%d%Y%H%M%S"),
+        help="Session name for the assume role.")
```
Comment (ashetkar) [inline aws/ec2/snappy_ec2.py]: "Name of this session in which this IAM role is assumed by the user."

**Review by ashetkar — COMMENTED** [review]
Please see the suggested changes in help text. Also, need to work with @lizygeogy to add this into documentation.

File: aws/ec2/snappy_ec2.py (hunk 3 above)
Comment (PradeepSurale) [reply aws/ec2/snappy_ec2.py]: yes time limit is 900-3600 enforced in AWS API itself.

**Pull request merged**